contracts/OptimisticArbitrator.sol

```diff
@@ -0,0 +1,200 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.17;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "@uma/core/contracts/oracle/interfaces/StoreInterface.sol";
+import "@uma/core/contracts/oracle/interfaces/FinderInterface.sol";
+import "@uma/core/contracts/oracle/interfaces/IdentifierWhitelistInterface.sol";
+import "@uma/core/contracts/oracle/interfaces/OptimisticOracleV2Interface.sol";
+
+import "@uma/core/contracts/common/implementation/AddressWhitelist.sol";
+import "@uma/core/contracts/oracle/implementation/Constants.sol";
+import "@uma/core/contracts/common/implementation/Testable.sol";
+
+contract OptimisticArbitrator is Testable {
+    using SafeERC20 for IERC20;
+
+    struct Request {
+        address proposer; // Address of the proposer.
+        address disputer; // Address of the disputer.
+        IERC20 currency; // ERC20 token used to pay rewards and fees.
+        bool settled; // True if the request is settled.
+        uint256[4] proposedPrice; // Price that the proposer submitted.
```

Comment:
as discussed we will change this to `proposedPrice` and talk about the ability to propose dynamic types in the tutorial.